app/models/deal.rb

```diff
@@ -0,0 +1,43 @@
+class Deal < ApplicationRecord
+  has_many :deal_images, dependent: :delete_all
+  accepts_nested_attributes_for :deal_images, allow_destroy: true, reject_if: proc { |attributes| attributes[:image].blank? }
+
+  validates :title, :description, :price_in_cents, :discount_price_in_cents, :quantity, :deals_tax, presence: true
+
+  validate :valid_publish_date?
+
+  validates :scheduled_deals_count, numericality: { less_than: 2, message: "No more than 2 deals can be published in one day" }, unless: :published_date
+
+  validates :discount_price_in_cents, numericality: { less_than_equal_to: :price_in_cents }
+
+  with_options if: :published_date? do
+    validates :quantity, numericality: { greater_than: 10, message: ' should be greater than 10' }
```

Comment:
make sure all the numbers are constants,example: MIN_DEAL_QUANTITY = 10

Reply:
done